document/CoreFormat.mdk

```diff
@@ -2225,3 +2260,112 @@ of the resulting UTF-8 byte sequence.
 
 * TODO: Relationship between original file encoding, offset, and UTF-8 byte sequence.
 * TODO: Explain responsibility of viewer/editor to match encoding.
+
+## `notification` object { #notification }
+
+### General { #notification-General }
+
+A `notification` object describes a condition encountered in the course of running an analysis tool
+which is relevant to the operation of the tool itself,
+as opposed to being relevant to a file being analyzed by the tool.
+Conditions relevant to files being analyzed by a tool are represented by `result` objects (&sect;[#result]).
+
+### `id` property { #notification-id }
+
+A `notification` object may contain a property named `id` whose value is a string containing
+the stable, opaque identifier for the condition that was encountered.
+
+### `ruleId` property { #notification-ruleId }
+
+If the condition described by the `notification` object is relevant to a particular analysis rule,
+the `notification` property should contain a property named `ruleId` whose value is the stable,
+unique identifier of the rule (&sect;[#rule-id]).
+
+### `message` property { #notification-message }
+
+A `notification` object shall contain a property named `message` whose value is a string that describes the condition that was encountered.
+
+### `level` property { #notification-level }
+
+A `notification` object may contain a property named `level` whose value is one of a fixed set of strings
+that specify the severity level of the notification.
+
+If present, the `level` property shall have one of the following values, with the specified meanings:
+
+* `error`: A serious problem was found.
```

Comment:
Do we want to provide more guidance? An error represents a condition where analysis halted or where results might be fundamentally compromised. a warning is appropriate when there's a lack of certainty around whether an issue occurred or where analysis may be incomplete but the results that were generated are probably valid. Warnings represent conditions that are either assumed to be transient or which really should be looked at (i.e., they are things which can and should be cleaned up in a perfect world of adequate resources). 'info' messages imply no requirement for action.